src/coreclr/src/pal/src/misc/jitsupport.cpp

```diff
@@ -13,6 +13,134 @@ SET_DEFAULT_DEBUG_CHANNEL(MISC);
 #include <asm/hwcap.h>
 #endif
 
+#if defined(HOST_ARM64) && defined(__linux__)
+typedef struct {
+    const char* name;
+    unsigned long hwCapFlag;
+} CpuCapability;
+
+static const CpuCapability CpuCapabilities[] = {
+    //{ "fp", HWCAP_FP },
+#ifdef HWCAP_ASIMD
+    { "asimd", HWCAP_ASIMD },
+#endif
+    //{ "evtstrm", HWCAP_EVTSTRM },
+#ifdef HWCAP_AES
+    { "aes", HWCAP_AES },
+#endif
+    //{ "pmull", HWCAP_PMULL },
+#ifdef HWCAP_SHA1
+    { "sha1", HWCAP_SHA1 },
+#endif
+#ifdef HWCAP_SHA2
+    { "sha2", HWCAP_SHA2 },
+#endif
+#ifdef HWCAP_CRC32
+    { "crc32", HWCAP_CRC32 },
+#endif
+#ifdef HWCAP_ATOMICS
+    { "atomics", HWCAP_ATOMICS },
+#endif
+    //{ "fphp", HWCAP_FPHP },
+    //{ "asimdhp", HWCAP_ASIMDHP },
+    //{ "cpuid", HWCAP_CPUID },
+#ifdef HWCAP_ASIMDRDM
+    { "asimdrdm", HWCAP_ASIMDRDM },
+#endif
+    //{ "jscvt", HWCAP_JSCVT },
+    //{ "fcma", HWCAP_FCMA },
+    //{ "lrcpc", HWCAP_LRCPC },
+    //{ "dcpop", HWCAP_DCPOP },
+    //{ "sha3", HWCAP_SHA3 },
+    //{ "sm3", HWCAP_SM3 },
+    //{ "sm4", HWCAP_SM4 },
+#ifdef HWCAP_ASIMDDP
+    { "asimddp", HWCAP_ASIMDDP },
+#endif
+    //{ "sha512", HWCAP_SHA512 },
+    //{ "sve", HWCAP_SVE },
+    //{ "asimdfhm", HWCAP_ASIMDFHM },
+    //{ "dit", HWCAP_DIT },
+    //{ "uscat", HWCAP_USCAT },
+    //{ "ilrcpc", HWCAP_ILRCPC },
+    //{ "flagm", HWCAP_FLAGM },
+    //{ "ssbs", HWCAP_SSBS },
+    //{ "sb", HWCAP_SB },
+    //{ "paca", HWCAP_PACA },
+    //{ "pacg", HWCAP_PACG },
+
+    // Ensure the array is never empty
+    { "", 0 }
+};
+
+// Returns the HWCAP_* flag corresponding to the given capability name.
+// If the capability name is not recognized or unused at present, zero is returned.
+static unsigned long LookupCpuCapabilityFlag(const char* start, size_t length)
+{
+    for (int i = 0; i < _countof(CpuCapabilities); i++)
+    {
+        const char* capabilityName = CpuCapabilities[i].name;
+        if ((length == strlen(capabilityName)) && (memcmp(start, capabilityName, length) == 0))
```

Comment:
>memcmp [](start = 51, length = 6)

curious, why `memcmp` is preferred than `strcmp`?

Reply:
@kunalspathak @jkotas I cannot use `strcmp`, because one of the strings is generally not zero-terminated but whitespace-terminated. And `strncmp(start, capabilityName, length) == 0` by itself — without the `length == strlen(capabilityName)` check — would not work in case `capabilityName` is longer than `length`. I could just replace `memcmp` with `strncmp` (keeping the first check), but that does not seem like a simplification to me.